src/tower_struggle/scenes/level_01.clj

```diff
@@ -20,11 +20,49 @@
     "img/big-background.png")
    (t/tetromino [300 800])])
 
+(defn tower-outline
+  [x y w h]
+  (let [tower-width w
+        tower-height h
+        coords {:tl  [x y]
+                :tr  [(+ x tower-width) y]
+                :bl  [x (+ y tower-height)]
+                :br  [(+ x tower-width) (+ y tower-height)]}
+        sides {:left   (concat (coords :tl) (coords :bl))
+               :right  (concat (coords :tr) (coords :br))
+               :top    (concat (coords :tl) (coords :tr))
+               :bottom (concat (coords :br) (coords :bl))}]
+    sides))
+
+(defn draw-tower-outline
+  "Draws a rectangle of dashed lines using Quil line primitives."
+  [x y w h]
+  (q/stroke common/black)
+  (let [o (tower-outline x y w h)
+        dash-length 10]
+    (doseq [[key value] o]
+      (let [distance (apply q/dist value)
+            dash-count (int (/ distance dash-length))
+            x1 (first value)
+            y1 (second value)
+            x2 (nth value 2)
+            y2 (nth value 3)
```

Comment:
these 4 bindings can be captured in a single destructuring of `value` (or `coords` once renamed).

```suggestion
            [x1 y1 x2 y2] coords
```